server/src/main/java/org/opensearch/querygroup/LabelingService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.querygroup;
+
+import org.opensearch.action.IndicesRequest;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.plugins.LabelingPlugin;
+
+import java.util.EnumMap;
+import java.util.List;
+
+/**
+ * TODO: Don't know the right package to put this class, need suggestions for maintainers on this
+ * Main class to hold multiple implementations of {@link org.opensearch.plugins.LabelingPlugin}
+ * this class will facilitate access and interactions to different implementations
+ * Usage: This class should be used as a member to orchestrate the working of your {@link LabelingPlugin}
+ */
+public class LabelingService {
+    /**
+     * Enum to define what all are currently implementing the plugin
+     */
+    public enum LabelingImplementationType {
```

Comment:
This should be moved to a separate file? Also are we suggesting a 1-1 mapping here? (a plugin can only be used for one certain purpose and one implementation can only be mapped to 1 plugin) Can multiple plugin have `QUERY_GROUP_RESOURCE_MANAGEMENT` type?

Reply:
Yes, I want this to be present since invoke triggers for feature might not be uniform, e,g; for QueryGrouping We will need to label the request at two places i,e; at co-ordinator request and at shard request. 
Given this will fall on the hot path it will be redundant re-compute the labels for all of the implementations

Regarding whether there should be a 1:1 mapping between feature and Implementation.
**Do you think it makes sense to utilise the same implementation across multiple features ? I am assuming each feature could be specific**

Reply:
I think it make sense to allow each feature to have their specific implementation.